src/app/ReadClient.cpp

```diff
@@ -1271,6 +1271,13 @@ CHIP_ERROR ReadClient::DefaultResubscribePolicy(CHIP_ERROR aTerminationCause)
         return CHIP_ERROR_LIT_SUBSCRIBE_INACTIVE_TIMEOUT;
     }
 
+    if (aTerminationCause.IsIMStatus() &&
+        aTerminationCause.GetSdkCode() == static_cast<uint8_t>(Protocols::InteractionModel::Status::InvalidAction))
```

Comment:
How about using ClusterStatusCode from a CHIP_ERROR: https://github.com/project-chip/connectedhomeip/blob/master/src/protocols/interaction_model/StatusCode.h#L71

That way the logic of what status is contained in an error is shared.